puppet-client.html

```diff
@@ -343,7 +343,34 @@
             this.fire("connectionerror", eventDetail);
 
             if (!eventDetail.handled) {
-                this._setConnectionError(JSON.stringify(eventDetail));
+                if (window.console && window.console.error) {
+                  var str = "Connection error: ";
+
+                  if (eventDetail.data.statusText) {
+                      str += eventDetail.data.statusText + " ";
```

Comment:
personally I'm a fan of giving labels in such cases. Like this:
```
status: <status>
method: <method>
url: <url>
```
That way it's not so confusing if one of those things ends up `undefined` or empty string.

But I'm totally fine if you disagree and leave it like that.